mloop/visualizations.py

```diff
@@ -1025,24 +1025,33 @@ def plot_hyperparameters_vs_run(self, parameter_subset=None):
         figure_counter += 1
         plt.figure(figure_counter)
         
-        artists=[]
-        for ind in range(num_params):
-            param_index = parameter_subset[ind]
-            color = param_colors[ind]
-            if self.num_params == 1:
-                plt.plot(self.fit_numbers,self.log_length_scale_history,'o',color=color)
-            else:
-                plt.plot(self.fit_numbers,self.log_length_scale_history[:,param_index],'o',color=color)
-            artists.append(plt.Line2D((0,1),(0,0), color=color,marker='o',linestyle=''))
-            
-        legend_labels = mlu._generate_legend_labels(
-            parameter_subset,
-            self.param_names,
-        )
-        plt.legend(artists, legend_labels ,loc=legend_loc)
+        if type(self.length_scale) is float:
+            # First treat the case of an isotropic kernel with one length scale
+            # shared by all parameters.
+            plt.plot(self.fit_numbers,self.log_length_scale_history,'o',color=param_colors[0])
+            plt.title('GP Learner: Log of length scale vs fit number.')
+        else:
+            # Now treat case of non-isotropic kernels with one length scale per
+            # parameter.
+            artists=[]
+            for ind in range(num_params):
+                param_index = parameter_subset[ind]
+                color = param_colors[ind]
+                if self.num_params == 1:
```

Comment:
Can we drop this check now? IIUC for a non-isotropic kernel with `num_params=1`, `log_length_scale_history` is still 2D.

Reply:
Good point. Actually when `num_params` is 1, theses lines
https://github.com/michaelhush/M-LOOP/blob/4916bedafb9ef51fe9420b5fe561fd4ceae4da04/mloop/learners.py#L1074-L1075
still convert `self.length_scale` to a float as if an isotropic kernel were specified. So this case ends up being dealt with by the first half of the `if` clause:
https://github.com/michaelhush/M-LOOP/blob/8605f3118987c69159b3f61ff74d17e95378ddd3/mloop/visualizations.py#L1028-L1032
That wasn't intended when I wrote it but is a bit of a happy accident. It means that the plot title gets the plurality right because it then says "Log of length scale" instead of "Log of length scales". That also means you're correct and the check that you mentioned can be removed, so I took it out in 8605f31.

Reply:
Hm, indeed, not sure how I missed that. Still, thanks for the change!